src/Generator/HtmlForPdfGenerator.php

```diff
@@ -2,41 +2,141 @@
 
 namespace SymfonyDocsBuilder\Generator;
 
+use Symfony\Component\DomCrawler\Crawler;
 use Symfony\Component\Filesystem\Filesystem;
 use Symfony\Component\Finder\Finder;
 use SymfonyDocsBuilder\BuildContext;
 
-/**
- * Class HtmlForPdfGenerator
- */
 class HtmlForPdfGenerator
 {
     use GeneratorTrait;
 
-    public function generateHtmlForPdf(
-        array $documents,
-        BuildContext $buildContext
-    ) {
+    public function generateHtmlForPdf(array $documents, BuildContext $buildContext) {
         $environments = $this->extractEnvironments($documents);
 
         $finder = new Finder();
         $finder->in($buildContext->getHtmlOutputDir())
             ->depth(0)
-            ->notName($buildContext->getParseOnly());
+            ->notName([$buildContext->getParseSubPath(), '_images']);
 
         $fs = new Filesystem();
         foreach ($finder as $file) {
             $fs->remove($file->getRealPath());
         }
 
-        $basePath  = sprintf('%s/%s', $buildContext->getHtmlOutputDir(), $buildContext->getParseOnly());
+        $basePath  = sprintf('%s/%s', $buildContext->getHtmlOutputDir(), $buildContext->getParseSubPath());
         $indexFile = sprintf('%s/%s', $basePath, 'index.html');
         if (!$fs->exists($indexFile)) {
-            throw new \InvalidArgumentException('File "%s" does not exist', $indexFile);
+            throw new \InvalidArgumentException(sprintf('File "%s" does not exist', $indexFile));
         }
 
+        // extracting all files from index's TOC, in the right order
         $parserFilename = $this->getParserFilename($indexFile, $buildContext->getHtmlOutputDir());
         $meta           = $this->getMeta($environments, $parserFilename);
-        dump(current($meta->getTocs()));
+        $files          = current($meta->getTocs());
+        array_unshift($files, sprintf('%s/index', $buildContext->getParseSubPath()));
+
+        // building one big html file with all contents
+        $content = '';
+        $htmlDir = $buildContext->getHtmlOutputDir();
+        $relativeImagesPath = str_repeat('../', substr_count($buildContext->getParseSubPath(), '/'));
+        foreach ($files as $file) {
+            $meta = $this->getMeta($environments, $file);
+
+            $filename = sprintf('%s/%s.html', $htmlDir, $file);
+            if (!$fs->exists($filename)) {
+                throw new \LogicException(sprintf('File "%s" does not exist', $filename));
+            }
+
+            // extract <body> content
+            $crawler     = new Crawler(file_get_contents($filename));
+            $fileContent = $crawler->filter('body')->html();
```

Comment:
A shame there's no way to fetch the built-content directory from `rst-parser`...

Reply:
actually because we're working from the already-generated html files, that seems impossible...
Or it would expose some services which does that work... but the `DomCrawler` already exists ;)

Reply:
Yea, like if rst-parser "stored" the final HTML somewhere (Meta is probably not the right place, but you get the idea) - and we could "query" for it from the Builder after - just something to think about for later.

Reply:
this was my first thought when i've started to do this feature... but html is never stored in any vars...